Traitify

Timidger opened the pull request:
Added traits to `WlcView` (`WlcViewable`) and `WlcOutput` (`WlcOutputable`). This allows other applications to make test harnesses by making fake handles for test suites.

This should bump us to `0.5`, so please add the appropriate tags and update the crate when you can Snirk.

Timidger committed:
Put the handle fn in traits, exported

Timidger committed:
WlcView & WlcOutput derive Copy trait

SnirkImmington commented:
Can we rename the traits to `Output` and `View`? I think those would be better names (and match Rust trait naming scheme, i.e. `Debug`, `Clone`, `Deref`, etc.).

Timidger commented:
Yeah that's fine. Will it conflict with the struct names though? And if so what do we name those?

SnirkImmington commented:
I've wanted to rename/alias them to `View` and `Output` since before we released on crates.io but if we can give the traits the better names that works.

How should we have `callbacks` handle the trait types? Surely one would want to initialize a test harness which sends a `FakeView` to the callbacks so they need to receive an `&View` or something.
Would wlc let us do that?

Also, we shouldn't we have the `View` and `Output` traits extend `Debug`, `Clone`, `PartialEq`, etc?

Timidger commented:
I don't think we need to worry about that case, since the fake views/outputs are only for testing so should really never interact with wlc in the first case. The use case I was envisioning was for testing code that relies on `View` and `Output` methods to return some data that would be normally queried from WLC but instead is overloaded by a function that returns some default. E.g:

```
fn get_geometry(&self) -> Geometry {
     Geometry {
        origin: Point { x: 0, y: 0},
        size: Size { w: 800, h: 600 }
     }
}
```

Yes, the traits should extend those particular traits since all of the structs that implement it should have those same methods

Timidger committed:
WlcOutputable -> Output, WlcViewable -> View

Renamed traits to the much more succinct Output and View instead of
WlcOutputable and WlcView

Timidger committed:
Added impl requirements to users of View/Output

Structs that implement View and Output now are required to implement the
following traits:
* Debug
* Clone
* Copy
* Eq/PartialEq
* Ord/PartialOrd
* Hash
